src/codemap.rs

```diff
@@ -94,7 +94,7 @@ impl LineRangeUtils for CodeMap {
         );
 
         LineRange {
-            file: lo.file.clone(),
+            file: Rc::clone(&lo.file),
```

Comment:
This seems like a bad change to me

Reply:
The rational behind this is (taken from clippy's document):

> #### What it does
>Checks for usage of .clone() on a ref-counted pointer, (Rc, Arc, rc::Weak, or sync::Weak), and suggests calling Clone on the corresponding trait instead.
>#### Why is this bad
>: Calling ‘.clone()’ on an Rc, Arc, or Weak can obscure the fact that only the pointer is being cloned, not the underlying data.

Reply:
I think Clippy is wrong here. I don't see why it is useful to selectively apply this rule, nor why there is a significant difference between cloning a smart pointer and some other data type (other data can be small too).

Reply:
Yeah just using a plain `.clone()` seems better. https://github.com/rust-lang-nursery/rustfmt/pull/2271/commits/516f15aba154e2cc66b999d0c9613479f4d53275 does not include this, so if there is no other problems, I think this PR can be merged.